src/game/Spell.cpp

```diff
@@ -2982,9 +2982,12 @@ void Spell::cast(bool skipCheck)
     // CAST SPELL
     SendSpellCooldown();
 
-    TakePower();
-    TakeReagents();                                         // we must remove reagents before HandleEffects to allow place crafted item in same slot
-    TakeAmmo();
+    if (!m_IsTriggeredSpell)
```

Comment:
I don't really rememeber but as @killerwife said triggered spells should cost no power hence is should belong to CalculatePowerCost -> will have the same result but in a better place.